src/workerd/api/tracing-module.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017-2025 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include "http.h"
+#include "worker-rpc.h"
+
+#include <workerd/io/io-context.h>
+
+namespace workerd::api {
+
+// Forward declaration for InternalSpanImpl
+class InternalSpanImpl;
+
+// Wrapper class that manages span ownership through IoContext or direct ownership
+class InternalSpan: public jsg::Object {
+ private:
+  kj::OneOf<kj::Own<InternalSpanImpl>, IoOwn<InternalSpanImpl>> impl;
+
+ public:
+  InternalSpan(kj::OneOf<kj::Own<InternalSpanImpl>, IoOwn<InternalSpanImpl>> impl);
+
+  void end();
+  void setTag(jsg::Lock& js, const kj::String key, const kj::Maybe<kj::OneOf<bool, double, kj::String>> value);
+  bool getIsRecording();
+  SpanParent makeSpanParent();
+
+  JSG_RESOURCE_TYPE(InternalSpan) {
+    JSG_METHOD(end);
+    JSG_METHOD(setTag);
+    JSG_METHOD(getIsRecording);
+  }
+
+  friend class InternalSpanImpl;
```

Comment:
Style-wise, we *typically* put the `public` section first. There are a handful of exceptions but this doesn't appear to qualify as one :-)

```suggestion
 public:
  InternalSpan(kj::OneOf<kj::Own<InternalSpanImpl>, IoOwn<InternalSpanImpl>> impl);

  void end();
  void setTag(jsg::Lock& js, const kj::String key, const kj::Maybe<kj::OneOf<bool, double, kj::String>> value);
  bool getIsRecording();
  SpanParent makeSpanParent();

  JSG_RESOURCE_TYPE(InternalSpan) {
    JSG_METHOD(end);
    JSG_METHOD(setTag);
    JSG_METHOD(getIsRecording);
  }

 private:
  kj::OneOf<kj::Own<InternalSpanImpl>, IoOwn<InternalSpanImpl>> impl;

  friend class InternalSpanImpl;
```